src/parser/core/modules/Gauge/CounterGauge.ts

```diff
@@ -148,7 +148,7 @@ export class CounterGauge extends AbstractGauge {
 	set(value: number, reason?: GaugeEventReason) {
 		const delta = value - this.value
 
-		const _reason = reason ?? delta > 0 ? 'generate' : 'spend'
+		const _reason = reason ?? (delta > 0 ? 'generate' : 'spend')
```

Comment:
I think the convention for an underscore prefix in a variable name is "we don't intend to use this variable for anything", but we are using it here. I also think the inline ternary operator is a little hard to parse (which is probably why this error got through in the first place). Could we change it to something like this instead?

```ts
    const defaultReason = (delta > 0) ? 'generate' : 'spend'
    
    // ...
    
    this.pushHistory(reason ?? defaultReason, delta)
```

Reply:
Honestly can't remember why I wrote it that way to begin with, so yes, happy to fix.